tests/components/vegehub/test_config_flow.py

```diff
@@ -0,0 +1,303 @@
+"""Tests for VegeHub config flow."""
+
+from ipaddress import ip_address
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components import zeroconf
+from homeassistant.components.vegehub.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+# Mock data for testing
+TEST_IP = "192.168.0.100"
+TEST_MAC = "A1:B2:C3:D4:E5:F6"
+TEST_SIMPLE_MAC = "A1B2C3D4E5F6"
+TEST_HOSTNAME = "VegeHub"
+
+DISCOVERY_INFO = zeroconf.ZeroconfServiceInfo(
+    ip_address=ip_address(TEST_IP),
+    ip_addresses=[ip_address(TEST_IP)],
+    port=80,
+    hostname=f"{TEST_HOSTNAME}.local.",
+    type="mock_type",
+    name="myVege",
+    properties={
+        zeroconf.ATTR_PROPERTIES_ID: TEST_HOSTNAME,
+        "version": "5.1.1",
+    },
+)
+
+
+@pytest.fixture
+def setup_mock_config_flow():
+    """Fixture to set up the mock config flow."""
+    with (
+        patch(
+            "socket.gethostname",
+            return_value=TEST_HOSTNAME,
+        ),
+    ):
+        yield
+
+
+async def test_user_flow_success(
+    hass: HomeAssistant, setup_mock_config_flow, mock_aiohttp_session
+) -> None:
+    """Test the user flow with successful configuration."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == FlowResultType.FORM
```

Comment:
```suggestion
    assert result["type"] is FlowResultType.FORM
```
everywhere you are comparing enums

Reply:
Fixed in 2c6235b5c7bb701c30c2553ee2b172ad97027e62.